SharedPackages/BrowserServicesKit/Sources/Subscription/SubscriptionUserScript/SubscriptionUserScript.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  SubscriptionUserScript.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Common
+import UserScript
+import WebKit
+
+///
+/// This protocol describes the interface for `SubscriptionUserScript` message handler
+///
+protocol SubscriptionUserScriptHandling {
+
+    /// Returns a handshake message reporting capabilities of the app.
+    func handshake(params: Any, message: UserScriptMessage) async throws -> SubscriptionUserScript.DataModel.HandshakeResponse
+
+    /// Returns the details of Privacy Pro subscription.
+    func subscriptionDetails(params: Any, message: UserScriptMessage) async throws -> SubscriptionUserScript.DataModel.SubscriptionDetails
+}
+
+final class SubscriptionUserScriptHandler: SubscriptionUserScriptHandling {
+    typealias DataModel = SubscriptionUserScript.DataModel
+
+    let platform: DataModel.Platform
+    let subscriptionManager: any SubscriptionAuthV1toV2Bridge
+
+    init(platform: DataModel.Platform, subscriptionManager: any SubscriptionAuthV1toV2Bridge) {
+        self.platform = platform
+        self.subscriptionManager = subscriptionManager
+    }
+
+    func handshake(params: Any, message: any UserScriptMessage) async throws -> DataModel.HandshakeResponse {
+        .init(availableMessages: [.subscriptionDetails], platform: platform)
+    }
+
+    func subscriptionDetails(params: Any, message: any UserScriptMessage) async throws -> DataModel.SubscriptionDetails {
+        guard let subscription = try? await subscriptionManager.getSubscription(cachePolicy: .returnCacheDataElseLoad), subscription.isActive else {
+            return .notSubscribed
+        }
+        return .init(subscription)
+    }
+}
+
+///
+/// This user script is responsible for providing Privacy Pro subscription data to the calling website.
+///
+public final class SubscriptionUserScript: NSObject, Subfeature {
+
+    public enum MessageName: String, CaseIterable, Codable {
+        case handshake
+        case subscriptionDetails
+    }
+
+    public let featureName: String = "subscriptions"
+    public let messageOriginPolicy: MessageOriginPolicy = .only(rules: [.exact(hostname: "duckduckgo.com")])
+    public weak var broker: UserScriptMessageBroker?
```

Comment:
Minor: Don’t think this is needed or used anywhere. (I logged `didSet` and it was never set)

Reply:
Great catch! This is indeed not needed unless we want to post messages from native to FE. Let me remove it :)

Reply:
Ah of course. It’s required by `Subfeature` protocol. So it’s going to stay :)